lib/ace/ext/papeeria/papeeria_latex_highlight_rules.coffee

```diff
@@ -95,13 +118,24 @@ define((require, exports, module) ->
       {
         token: [
           "storage.type"
+          "lparen." + ENVIRONMENT_TOKENTYPE
+          "variable.parameter." + ENVIRONMENT_TOKENTYPE
+          "rparen"
           "lparen"
           "variable.parameter"
           "rparen"
         ]
-        regex: "(\\\\(?:begin|end))({)(\\w*)(})"
+        regex: "(\\\\(?:begin|end))({)(tabular)(})({)(.*)(})"
+      }
+      {
+        token: [
+          "storage.type"
+          "lparen.environment." + ENVIRONMENT_TOKENTYPE
+          "variable.parameter." + ENVIRONMENT_TOKENTYPE
+          "rparen"
+        ]
+        regex: "(\\\\(?:begin|end))({)(.*)(})"
```

Comment:
out of curiosity: are all these capture groups in the regexp used? Can we remove parentheses around `{` and `}`?